strava_oauth2 returns inline

AartGoossens opened the pull request:
The access token from Strava is currently returned in the browser, not inline in your Python code.
I had an idea on how to return the access token inline in your Python code like this:
```python
from stravaio import strava_oauth2
token = strava_oauth2(client_id=STRAVA_CLIENT_ID, client_secret=STRAVA_CLIENT_SECRET)
```
This PR includes the code that adds this feature. The code is a bit wonky (e.g. too much use of low level `socket` and `urllib` thingies), but I hope you are open to accepting it!

By the way: The returned token object is the full Strava response for the OAuth2 token exchange, which includes not only the access token, but also the refresh token among other things. I think it might make sense to e.g. only return access and refresh token.

AartGoossens committed:
strava_oauth2 returns inline

sladkovm commented:
Nice! It was my original intention to implement this exact functionality but I could not come up with a working 'wonky' code ;)

I have trouble to run it on Windows10:

```
  File "c:\users\sladkov\documents\private\projects\stravaio\stravaio.py", line 380, in run_server_and_wait_for_token
    chunk = conn.recvmsg(16)
AttributeError: 'socket' object has no attribute 'recvmsg'
```

AartGoossens commented:
Gotcha: Only just noticed the "Availability: most Unix platforms, possibly others." in the Python [docs](https://docs.python.org/3.6/library/socket.html#socket.socket.recvmsg).
I changed it to use `recv()` which is apparently available on all platforms and updated the PR.

AartGoossens commented:
:man_dancing:
